### PR TITLE
fix docs for decoder_input_ids

### LINKED ISSUE #11357: possible mistake in documentation
Looking at description of the parameter "decoder_input_ids" in "forward" method of BartForConditionalGeneration/T5ForConditionalGeneration, I see following:

BartForConditionalGeneration:
decoder_input_ids - ... For translation and summarization training, decoder_input_ids should be provided. If no decoder_input_ids is provided, the model will create this tensor by shifting the !!INPUT_IDS!! to the right for denoising pretraining following the paper.

T5ForConditionalGeneration:
decoder_input_ids - ... To know more on how to prepare decoder_input_ids for pretraining take a look at T5 Training. If decoder_input_ids and decoder_inputs_embeds are both unset, decoder_input_ids takes the value of  !!INPUT_IDS!!.

Looks like there should be LABELS instead of INPUT_IDS.

Thanks,
@patrickvonplaten, @patil-suraj

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

Few doc fixes for `decoder_input_ids` in s2s models.

Fixes #11357

Thanks, @shyrma  for spotting this!

### CODE REVIEW

File: src/transformers/models/bart/modeling_bart.py [hunk 1]
```diff
@@ -582,14 +582,14 @@ def __init_subclass__(self):
...
 
-            `What are input IDs? <../glossary.html#input-ids>`__
+            `What are decoder input IDs? <../glossary.html#decoder-input-ids>`__
 
             Bart uses the :obj:`eos_token_id` as the starting token for :obj:`decoder_input_ids` generation. If
             :obj:`past_key_values` is used, optionally only the last :obj:`decoder_input_ids` have to be input (see
             :obj:`past_key_values`).
 
             For translation and summarization training, :obj:`decoder_input_ids` should be provided. If no
-            :obj:`decoder_input_ids` is provided, the model will create this tensor by shifting the :obj:`input_ids` to
```
Comment (patrickvonplaten) [inline src/transformers/models/bart/modeling_bart.py:592]: this was actually correct - for BART if no `input_ids` **and** no `labels` are provided -> then `decoder_input_ids` are shifted according to the `input_ids` => this corresponds to BART's denoising pre-training

File: src/transformers/models/bart/modeling_tf_bart.py [hunk 2]
```diff
@@ -559,14 +559,14 @@ def serving(self, inputs):
...
-            `What are input IDs? <../glossary.html#input-ids>`__
+            `What are decoder input IDs? <../glossary.html#decoder-input-ids>`__
 
             Bart uses the :obj:`eos_token_id` as the starting token for :obj:`decoder_input_ids` generation. If
             :obj:`past_key_values` is used, optionally only the last :obj:`decoder_input_ids` have to be input (see
             :obj:`past_key_values`).
 
             For translation and summarization training, :obj:`decoder_input_ids` should be provided. If no
-            :obj:`decoder_input_ids` is provided, the model will create this tensor by shifting the :obj:`input_ids` to
+            :obj:`decoder_input_ids` is provided, the model will create this tensor by shifting the :obj:`labels` to
```
Comment (patrickvonplaten) [inline src/transformers/models/bart/modeling_tf_bart.py:569]: this was actually correct - for BART if no `input_ids` **and** no `labels` are provided -> then `decoder_input_ids` are shifted according to the `input_ids` => this corresponds to BART's denoising pre-training

File: src/transformers/models/mbart/modeling_mbart.py [hunk 3]
```diff
@@ -582,15 +582,15 @@ def dummy_inputs(self):
...
+            `What are decoder input IDs? <../glossary.html#decoder-input-ids>`__
 
             MBart uses a specific language id token as the starting token for :obj:`decoder_input_ids` generation that
             varies according to source and target language, *e.g.* 25004 for `en_XX`, and 25003 for `de_DE`. If
             :obj:`past_key_values` is used, optionally only the last :obj:`decoder_input_ids` have to be input (see
             :obj:`past_key_values`).
 
             For translation and summarization training, :obj:`decoder_input_ids` should be provided. If no
-            :obj:`decoder_input_ids` is provided, the model will create this tensor by shifting the :obj:`input_ids` to
+            :obj:`decoder_input_ids` is provided, the model will create this tensor by shifting the :obj:`labels` to
```
Comment (patrickvonplaten) [inline src/transformers/models/mbart/modeling_mbart.py:593]: this was actually correct - for mBART if no `input_ids` **and** no `labels` are provided -> then `decoder_input_ids` are shifted according to the `input_ids` => this corresponds to mBART's denoising pre-training

File: src/transformers/models/mbart/modeling_tf_mbart.py [hunk 4]
```diff
@@ -533,15 +533,15 @@ def serving(self, inputs):
...
+            `What are decoder input IDs? <../glossary.html#decoder-input-ids>`__
 
             MBart uses a specific language id token as the starting token for :obj:`decoder_input_ids` generation that
             varies according to source and target language, *e.g.* 25004 for `en_XX`, and 25003 for `de_DE`. If
             :obj:`past_key_values` is used, optionally only the last :obj:`decoder_input_ids` have to be input (see
             :obj:`past_key_values`).
 
             For translation and summarization training, :obj:`decoder_input_ids` should be provided. If no
-            :obj:`decoder_input_ids` is provided, the model will create this tensor by shifting the :obj:`input_ids` to
+            :obj:`decoder_input_ids` is provided, the model will create this tensor by shifting the :obj:`labels` to
```
Comment (patrickvonplaten) [inline src/transformers/models/mbart/modeling_tf_mbart.py:544]: this was actually correct - for mBART if no `input_ids` **and** no `labels` are provided -> then `decoder_input_ids` are shifted according to the `input_ids` => this corresponds to mBART's denoising pre-training

File: src/transformers/models/bart/modeling_bart.py [hunk 5]
```diff
@@ -582,14 +582,14 @@ def __init_subclass__(self):
             :meth:`transformers.PreTrainedTokenizer.encode` and :meth:`transformers.PreTrainedTokenizer.__call__` for
             details.
 
-            `What are input IDs? <../glossary.html#input-ids>`__
+            `What are decoder input IDs? <../glossary.html#decoder-input-ids>`__
```
Comment (patrickvonplaten) [inline src/transformers/models/bart/modeling_bart.py:585]: Thanks for correcting this!

**Comment by patil-suraj** [comment]
Thanks a lot for catching this Patrick! I corrected this for BART and mBART.

**Review by patrickvonplaten — APPROVED** [review]
